File: LeetCode/66.java
//Plus One
class Solution {
    public int[] plusOne(int[] digits) {
        int len = digits.length;
        int n = 0, cnt = 0;
        
        for(int i = 0; i < len; i++){
            n += digits[i] * pow(10, 2 - i);
            
        }
        n++;
        int tmp = n;
        while(tmp != 0){
            tmp /= 10;
            cnt++;
        }
        for (int i = 0; i < cnt; i++){
            digits[cnt - 1 - i] = n % 10;
            n /= 10;
        }
        return digits;
        
    }
    public static int pow(int base, int exp){
        int n = 1;
        for(int i = 0; i < exp; i++){
            n *= base;
        }
        return base;
    }
}

File: LeetCode/66.cpp
#include <algorithm>
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int last_index = digits.size() - 1;
        while(last_index >= 0 && digits[last_index] == 9){
            digits[last_index--] = 0;
            
        }
        if(last_index == -1){
            digits.push_back(1);
            reverse(digits.begin(), digits.end());
        }
        else digits[last_index]++;
        return digits;
    }
};